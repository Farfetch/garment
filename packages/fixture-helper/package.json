{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.16.0",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.16.0",
    "@garment/runner": "^0.16.0",
    "@garment/workspace": "^0.16.0",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  },
  "devDependencies": {
    "is-valid-path": "^0.1.1",
    "normalize-path": "^3.0.0"
  }
}
