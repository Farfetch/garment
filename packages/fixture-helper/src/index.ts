export * from './executeRunner';
export * from './initFixtureHelper';
export * from './resolver';
import normalizePath = require('normalize-path');
import isValidPath = require('is-valid-path');

const containsValidPath = (value: string): boolean =>
  value.split(/\s/).some((s) => isValidPath(s));

export const normalizeContainedPath = (value: string): string => {
  return value
    .split(/(\s)/)
    .map((s) => {
      if (isValidPath(s)) {
        return normalizePath(s);
      } else {
        return s;
      }
    })
    .join('');
};

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  const normalizedTestPathRegEx = new RegExp(normalizePath(testPath), 'g');
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string' && containsValidPath(value)
        ? normalizeContainedPath(value).replace(
            normalizedTestPathRegEx,
            replaceWith
          )
        : value
    )
  );
}
