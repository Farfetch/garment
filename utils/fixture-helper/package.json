{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.13.7",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.13.7",
    "@garment/runner": "^0.13.7",
    "@garment/workspace": "^0.13.7",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  }
}
