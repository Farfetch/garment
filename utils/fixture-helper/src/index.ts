export * from './executeRunner';
export * from './initFixtureHelper';

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string'
        ? value.replace(new RegExp(testPath, 'g'), replaceWith)
        : value
    )
  );
}
