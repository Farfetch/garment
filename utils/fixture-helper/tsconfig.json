{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "composite": true
  },
  "exclude": [
    "node_modules",
    "lib",
    "__tests__"
  ],
  "references": [
    {
      "path": "../../core/logger"
    },
    {
      "path": "../../core/runner"
    },
    {
      "path": "../../core/workspace"
    }
  ]
}