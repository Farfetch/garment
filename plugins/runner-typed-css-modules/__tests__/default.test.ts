import { executeRunner2, initFixtureHelper } from '@garment/fixture-helper';
import handler from '../src';

const { initFixture, clean } = initFixtureHelper(module, {
  tempDir: __dirname + 'tmp__'
});

afterAll(clean);

test('Creates typings for CSS file', async () => {
  const result = await executeRunner2(
    handler,
    {},
    {
      projectRelativePath: 'test-project',
      files: file => [
        file.text(
          'index.css',
          '.foo { color: red; } .bar { background: black; }'
        ),
        file.text('component/style.css', '.content { font-size: 2em; }')
      ]
    }
  );

  expect(result).toMatchSnapshot();
});
