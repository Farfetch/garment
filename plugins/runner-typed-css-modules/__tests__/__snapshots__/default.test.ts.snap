// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates typings for CSS file 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "data": "declare const styles: {
  readonly \\"foo\\": string;
  readonly \\"bar\\": string;
};
export = styles;

",
      "path": "index.css.d.ts",
      "type": "text",
    },
    Object {
      "data": "declare const styles: {
  readonly \\"content\\": string;
};
export = styles;

",
      "path": "component/style.css.d.ts",
      "type": "text",
    },
  ],
  "dependsOnFile": Array [],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Generating types for CSS modules",
      ],
      "level": "info",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;
